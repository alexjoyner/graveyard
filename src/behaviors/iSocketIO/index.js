import { env } from "../../.env";
const io = require('socket.io-client');
const socket = io(env.serverAddr);

class iSocketIO {
  constructor(){
    if(this.subscribe === undefined)
      throw new TypeError("Must override method");
    if(this.unsubscribe === undefined)
      throw new TypeError("Must override method");
    if(this.join === undefined)
      throw new TypeError("Must override method");
  }
}

class PointsSocket extends iSocketIO {
  constructor(){super()}
  subscribe(pointsArray, cb){
    this.join(pointsArray);
    socket.on('add log', (log) => {
        cb(null, log);
    })
  }
  unsubscribe(pointsArray, cb){
    socket.emit('leave-group', pointsArray);
  }
  join(pointsArray){
    socket.emit('join-group', pointsArray);
  }
}


export {PointsSocket};