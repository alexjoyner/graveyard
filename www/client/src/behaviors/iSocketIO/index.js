import io from 'socket.io-client';
import { env } from '../../.env';

export class ISocketIO {
  constructor() {
    if (this.subscribe === undefined) { throw new TypeError('Must override method'); }
    if (this.unsubscribe === undefined) { throw new TypeError('Must override method'); }
    if (this.join === undefined) { throw new TypeError('Must override method'); }
  }
}

class PointsSocket extends ISocketIO {
  constructor(opts) {
    super();
    this.socket = opts.socket || /* istanbul ignore next */ io(env.serverAddr);
  }
  subscribe(pointsArray, cb) {
    this.points = pointsArray;
    this.join();
    this.socket.on('add log', /* istanbul ignore next */ (log) => {
      cb(null, log);
    });
  }
  unsubscribe() {
    this.socket.emit('leave-group', this.points);
  }
  join() {
    this.socket.emit('join-group', this.points);
  }
  setPoints(pointsArray) {
    this.points = pointsArray;
  }
  getPoints() {
    return this.points;
  }
}

export { PointsSocket };
