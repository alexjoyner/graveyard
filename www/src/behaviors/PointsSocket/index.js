import io from 'socket.io-client';
import { env } from '../../.env';
import { ISocketIO } from '../interfaces/ISocketIO';

class PointsSocket extends ISocketIO {
  constructor() {
    super();
    this.points = [];
    const connectionOptions = {
      'force new connection': true,
      reconnectionAttempts: 'Infinity', // avoid having user reconnect manually in order to prevent dead clients after a server restart
      timeout: 10000, // before connect_error and connect_timeout are emitted.
      transports: ['websocket'],
    };
    this.socket = io(env.serverAddr, connectionOptions);
  }

  subscribe(pointsArray, cb) {
    this.points = pointsArray;
    this.socket.on('add log', /* istanbul ignore next */ (log) => {
      cb(null, log);
    });
    this.join();
    this.getLastLogs();
  }

  unsubscribe() {
    if (this.points.length > 0) {
      this.socket.emit('leave-group', this.points);
    }
  }

  join() {
    this.socket.emit('join-group', this.points);
  }

  getLastLogs() {
    this.socket.emit('get-last-logs', this.points);
  }

  setPoints(pointsArray) {
    this.points = pointsArray;
  }

  getPoints() {
    return this.points;
  }
}

export { PointsSocket };
