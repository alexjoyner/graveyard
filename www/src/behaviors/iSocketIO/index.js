import io from 'socket.io-client';
import { env } from '../../.env';

export class ISocketIO {
  constructor() {
    if (this.subscribe === undefined) { throw new TypeError('Must override method'); }
    if (this.unsubscribe === undefined) { throw new TypeError('Must override method'); }
    if (this.join === undefined) { throw new TypeError('Must override method'); }
  }
}

class PointsSocket extends ISocketIO {
  constructor(opts) {
    super();
    this.socket = opts.socket || /* istanbul ignore next */ io(env.serverAddr);
  }

  subscribe(pointsArray, cb) {
    this.points = pointsArray;
    this.socket.on('add log', /* istanbul ignore next */ (log) => {
      cb(null, log);
    });
    this.join();
    this.getLastLogs();
  }

  unsubscribe() {
    this.socket.emit('leave-group', this.points);
  }

  join() {
    this.socket.emit('join-group', this.points);
  }

  getLastLogs() {
    this.socket.emit('get-last-logs', this.points);
  }

  setPoints(pointsArray) {
    this.points = pointsArray;
  }

  getPoints() {
    return this.points;
  }
}

export { PointsSocket };
