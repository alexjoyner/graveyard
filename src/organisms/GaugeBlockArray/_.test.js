import React from 'react';
import io from 'socket.io-client';
import { shallow } from 'enzyme';
import { BaseGaugeBlockArray } from './';
import { PointsSocket } from '../../behaviors/iSocketIO';

const mockFetch = data => jest.fn().mockImplementation(() => Promise.resolve({
  status: 200,
  json: () => data,
}));

jest.mock('socket.io-client', () => ({
  on: jest.fn(),
  emit: jest.fn(),
}));

window.fetch = mockFetch([
  { id: 1, name: 'testPoint' },
  { id: 2, name: 'testPoint2' },
]);

describe('BaseGaugeBlockArray component', () => {
  describe('Component without data', () => {
    let component;
    beforeEach(() => {
      const props = {
        publishNewPoints: jest.fn(),
        publishNewLog: jest.fn(),
        removeAllPoints: jest.fn(),
        socket: new PointsSocket({
          socket: io,
        }),
        points: {},
      };
      component = shallow(<BaseGaugeBlockArray {...props} />);
    });
    it('Should render without exploding', () => {
      expect(component).toBeDefined();
    });
    it('Should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
  describe('Component with points', () => {
    let component;
    beforeEach(() => {
      const props = {
        publishNewPoints: jest.fn(),
        publishNewLog: jest.fn(),
        removeAllPoints: jest.fn(),
        socket: new PointsSocket({
          socket: io,
        }),
        points: {
          1: { name: 'Test Input', value: 10 },
        },
      };
      component = shallow(<BaseGaugeBlockArray {...props} />);
    });
    it('Should render without exploding', () => {
      expect(component).toBeDefined();
    });
    it('Should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
