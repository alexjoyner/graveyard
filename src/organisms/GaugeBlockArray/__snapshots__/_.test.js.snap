// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseGaugeBlockArray component Component with points Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BaseGaugeBlockArray
    points={
        Object {
            "1": Object {
              "name": "Test Input",
              "value": 10,
            },
          }
    }
    publishNewLog={[Function]}
    publishNewPoints={[Function]}
    removeAllPoints={[Function]}
    socket={
        PointsSocket {
            "points": Array [
              "1",
              "2",
            ],
            "socket": Object {
              "emit": [Function],
              "on": [Function],
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Connect(Component)
          id="1"
          point={
                    Object {
                              "name": "Test Input",
                              "value": 10,
                            }
          }
          points={
                    Object {
                              "1": Object {
                                "name": "Test Input",
                                "value": 10,
                              },
                            }
          }
          publishNewLog={[Function]}
          publishNewPoints={[Function]}
          removeAllPoints={[Function]}
          socket={
                    PointsSocket {
                              "points": Array [
                                "1",
                                "2",
                              ],
                              "socket": Object {
                                "emit": [Function],
                                "on": [Function],
                              },
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "id": "1",
          "point": Object {
            "name": "Test Input",
            "value": 10,
          },
          "points": Object {
            "1": Object {
              "name": "Test Input",
              "value": 10,
            },
          },
          "publishNewLog": [Function],
          "publishNewPoints": [Function],
          "removeAllPoints": [Function],
          "socket": PointsSocket {
            "points": Array [
              "1",
              "2",
            ],
            "socket": Object {
              "emit": [Function],
              "on": [Function],
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(Component)
            id="1"
            point={
                        Object {
                                    "name": "Test Input",
                                    "value": 10,
                                  }
            }
            points={
                        Object {
                                    "1": Object {
                                      "name": "Test Input",
                                      "value": 10,
                                    },
                                  }
            }
            publishNewLog={[Function]}
            publishNewPoints={[Function]}
            removeAllPoints={[Function]}
            socket={
                        PointsSocket {
                                    "points": Array [
                                      "1",
                                      "2",
                                    ],
                                    "socket": Object {
                                      "emit": [Function],
                                      "on": [Function],
                                    },
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "id": "1",
            "point": Object {
              "name": "Test Input",
              "value": 10,
            },
            "points": Object {
              "1": Object {
                "name": "Test Input",
                "value": 10,
              },
            },
            "publishNewLog": [Function],
            "publishNewPoints": [Function],
            "removeAllPoints": [Function],
            "socket": PointsSocket {
              "points": Array [
                "1",
                "2",
              ],
              "socket": Object {
                "emit": [Function],
                "on": [Function],
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`BaseGaugeBlockArray component Component without data Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BaseGaugeBlockArray
    points={Object {}}
    publishNewLog={[Function]}
    publishNewPoints={[Function]}
    removeAllPoints={[Function]}
    socket={
        PointsSocket {
            "points": Array [
              "1",
              "2",
            ],
            "socket": Object {
              "emit": [Function],
              "on": [Function],
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
